src/node/miner.cpp

```diff
@@ -586,4 +586,43 @@ std::optional<BlockRef> WaitTipChanged(ChainstateManager& chainman, KernelNotifi
     // avoid deadlocks.
     return GetTip(chainman);
 }
+
+node::CoinbaseTemplate ExtractCoinbaseTemplate(const CTransactionRef coinbase_tx)
+{
+    node::CoinbaseTemplate coinbase;
+
+    coinbase.version = coinbase_tx->CURRENT_VERSION;
+    coinbase.script_sig_prefix = coinbase_tx->vin[0].scriptSig;
+    // The CoinbaseTemplate interface guarantees a size limit. Raising it (e.g.
+    // if a future softfork needs to commit more than BIP34) is a
+    // breaking change for clients.
+    Assume(coinbase.script_sig_prefix.size() <= 8);
```

Comment:
In commit "mining: add getCoinbase()" (80acabaf9353245f0839b18d9552ab39f595eda6)

It seems like if this change could break clients, it might be better to return an error than a template that violates their expectations. Maybe would suggest:

```c++
if (!Assume(coinbase.script_sig_prefix.size() <= 8)) {
  throw std::logic_error("Internal bug: coinbase data is inconsistent with CoinbaseTemplate definition. Data may be corrupt or definition may need to be updated");
}
```

Or could replace Assume with Assert, or log a warning. it just doesn't seem good to ignore the condition completely if it happens.